sw/device/lib/dif/autogen/dif_adc_ctrl_autogen.c

```diff
@@ -25,6 +25,18 @@ dif_result_t dif_adc_ctrl_init(mmio_region_t base_addr,
   return kDifOk;
 }
 
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_adc_ctrl_init_dt(const dt_adc_ctrl_t *dt,
```

Comment:
This function is doing exactly the same as `dif_adc_ctrl_init`, I wonder if `dif_adc_ctrl_init` should be removed to avoid confusion.

Reply:
we could also call this function `dif_adc_ctrl_from_dt`

Reply:
We need to keep both because we cannot change all tests are once and we don't want to break existing code. There is a comment in the autogen header:
>  DEPRECATED This function exists solely for the transition to dt-based DIFs and will be removed in the future.

I think you renaming suggestion is good.

Reply:
I have renamed it to `dif_adc_ctrl_init_from_dt`. I think it's important to keep the `init` to make it clear that this is creating a DIF.